src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -43,24 +43,14 @@ private enum FrameType : byte
             AppData = 23
         }
 
-        //
-        // This block is used to rule the >>re-handshakes<< that are concurrent with read/write I/O requests.
-        //
-        private const int LockNone = 0;
-        private const int LockWrite = 1;
-        private const int LockHandshake = 2;
-        private const int LockPendingWrite = 3;
-        private const int LockRead = 4;
-        private const int LockPendingRead = 6;
+        private object _handshakeLock = new object();
```

Comment:
```suggestion
        private readonly object _handshakeLock = new object();
```

Are there any other objects SslStream always creates and doesn't share out that we could use for this purpose instead?

Reply:
yes, I was thinking about using _context but in the end I decided to keep it separate for now. I can do more testing and I think it should be possible.

Reply:
I'm fine if you want to go with a dedicated object for now.  We can revisit it when the dust settles a little.